promql/engine.go

```diff
@@ -651,57 +653,133 @@ func (ng *Engine) populateSeries(ctx context.Context, querier storage.Querier, s
 		warnings  storage.Warnings
 		err       error
 	)
+	RemoteSelectCalls := 0
+	if querier.Remotely() {
```

Comment:
PromQL shouldn't care about how the querier is implemented, it should always be the same logic.

Reply:
thanks for your review.
I move "remotely" to the engine. Is that ok? Only three files need to be changed.
![image](https://user-images.githubusercontent.com/9583245/78540450-c35d8d80-7826-11ea-92e4-3becaf19a08f.png)
![image](https://user-images.githubusercontent.com/9583245/78540485-d1131300-7826-11ea-89bb-06d9e73aa86b.png)
![image](https://user-images.githubusercontent.com/9583245/78540393-ad4fcd00-7826-11ea-8393-08ffb855a9e3.png)

Reply:
That sort of check shouldn't appear anywhere in the promql directory, any special handling of local vs remote should be in fanout.

Reply:
Thanks, I will think about how to implement this feature in "fanout.go"

Reply:
I have moved the remote related logic to the fanout directory.
done . please review again ,thanks.
![image](https://user-images.githubusercontent.com/9583245/78564583-3aa71780-784f-11ea-8102-22c5e98e28c2.png)
debug info 9010 is remotely result, 9090 is locally result
![image](https://user-images.githubusercontent.com/9583245/78564692-69bd8900-784f-11ea-8fb9-63ebd050c4a8.png)